backend/src/main/java/com/klm/dev/exercises/devcase02/flight/FlightClient.java

```diff
@@ -0,0 +1,82 @@
+package com.klm.dev.exercises.devcase02.flight;
+
+import com.klm.dev.exercises.devcase02.weather.Weather;
+import com.klm.dev.exercises.devcase02.weather.WeatherClient;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@Service
+@Configuration
+@PropertySource("classpath:Flight.properties")
+public class FlightClient {
+    @Autowired
+    private RestTemplate restTemplate;
+
+    private static final int NTHREDS = 20;
+
+    @Autowired
+    private WeatherClient weatherClient;
+
+    @Value("${flights.url}")
+    private String url;
+//Goes through all flights one by one, and calls the weather information on all locations
+// per flight
+    public Flight[] getFlights() {
```

Comment:
Rename FlightClient to FlightService introduce a FlightClient which contains only the logic for retrieving like you did for weather. And only do the mapping(setting the weather information in the flight)